pkg/cmd/issue/view/presentation_issue.go

```diff
@@ -0,0 +1,122 @@
+	Reactions      string
+	AssigneesList  string
+	LabelsList     string
+	ProjectsList   string
+	MilestoneTitle string
+	Body           string
+	URL            string
+}
+
+// Creates a new PresentationIssue from an api.Issue.
+// The PresentationIssue is what the issue printers need to display an
+// issue to the user.
+func MapApiIssueToPresentationIssue(issue *api.Issue, colorScheme *iostreams.ColorScheme) (PresentationIssue, error) {
+	presentationIssue := PresentationIssue{
+		Title:         issue.Title,
+		Number:        issue.Number,
+		CreatedAt:     issue.CreatedAt,
+		Comments:      issue.Comments,
+		Author:        issue.Author.Login,
+		State:         issue.State,
+		StateReason:   issue.StateReason,
+		Reactions:     prShared.ReactionGroupList(issue.ReactionGroups),
+		AssigneesList: stringifyAssignees(issue.Assignees),
+		LabelsList:    stringifyAndColorizeLabels(sortAlphabeticallyIgnoreCase(issue.Labels), colorScheme),
+		ProjectsList:  stringifyProjects(issue.ProjectCards, issue.ProjectItems),
+		Body:          issue.Body,
+		URL:           issue.URL,
+	}
+
+	if issue.Milestone != nil {
+		presentationIssue.MilestoneTitle = issue.Milestone.Title
+	}
+
+	return presentationIssue, nil
+}
+
+func stringifyProjects(projectCards api.ProjectCards, projectItems api.ProjectItems) string {
+	if len(projectCards.Nodes) == 0 && len(projectItems.Nodes) == 0 {
+		return ""
+	}
+
+	projectNames := make([]string, len(projectCards.Nodes)+len(projectItems.Nodes))
+	for i, project := range projectCards.Nodes {
+		colName := project.Column.Name
+		if colName == "" {
+			colName = "Awaiting triage"
+		}
+		projectNames[i] = fmt.Sprintf("%s (%s)", project.Project.Name, colName)
+	}
+
+	for i, project := range projectItems.Nodes {
+		statusName := project.Status.Name
+		if statusName == "" {
+			statusName = "Backlog"
+		}
+		projectNames[i+len(projectCards.Nodes)] = fmt.Sprintf("%s (%s)", project.Project.Title, statusName)
+	}
+
+	list := strings.Join(projectNames, ", ")
+	if projectCards.TotalCount > len(projectCards.Nodes) {
+		list += ", …"
+	}
+	return list
+}
+
+func stringifyAssignees(issueAssignees api.Assignees) string {
+	if len(issueAssignees.Nodes) == 0 {
+		return ""
+	}
+
+	AssigneeNames := make([]string, 0, len(issueAssignees.Nodes))
+	for _, assignee := range issueAssignees.Nodes {
+		AssigneeNames = append(AssigneeNames, assignee.Login)
+	}
+
+	list := strings.Join(AssigneeNames, ", ")
+	if issueAssignees.TotalCount > len(issueAssignees.Nodes) {
+		list += ", …"
+	}
+	return list
+}
+
+func stringifyAndColorizeLabels(issueLabels api.Labels, colorScheme *iostreams.ColorScheme) string {
+	labelNames := make([]string, len(issueLabels.Nodes))
+	for j, label := range issueLabels.Nodes {
+		if colorScheme == nil {
+			labelNames[j] = label.Name
+		} else {
+			labelNames[j] = colorScheme.HexToRGB(label.Color, label.Name)
+		}
+	}
+
+	return strings.Join(labelNames, ", ")
+}
+
+func sortAlphabeticallyIgnoreCase(l api.Labels) api.Labels {
```

Comment:
I think this function signature is a bit confusing given the implementation.

```
sortAlphabeticallyIgnoreCase(l api.Labels) api.Labels
```

This indicates that I give it a labels and it gives me back sorted labels. This is true but it's kind of hiding the fact that the slice of Nodes is being mutated in place. This is also surprising in the way we use it because it's mutating the original `api.Issue` as well.